staging/src/k8s.io/apimachinery/pkg/util/httpstream/wsstream/conn.go

```diff
@@ -272,6 +277,46 @@ func (conn *Conn) handle(ws *websocket.Conn) {
 	}
 }
 
+// handleV5 handles websocket data according to "V5" of the remotecommand subprotocol. The
+// V5 protocol defines a RESET signal as which is transmitted in the first byte. RESET will
+// close the stream, allowing other streams to finish transmitting.
+func (conn *Conn) handleV5(ws *websocket.Conn) {
+	defer conn.Close()
+	conn.initialize(ws)
+
+	for {
+		conn.resetTimeout()
+		var data []byte
+		if err := websocket.Message.Receive(ws, &data); err != nil {
+			if err != io.EOF {
+				klog.Errorf("Error on socket receive: %v", err)
+			}
+			break
+		}
+		if len(data) == 0 {
+			continue
+		}
+		channel := data[0]
+		data = data[1:]
+		if channel == remotecommand.StreamReset {
+			if len(data) > 0 {
+				channel := data[0]
+				klog.V(4).Infof("Received half-close signal from client; close %d stream", channel)
+				conn.channels[channel].Close()
+			}
+			continue
+		}
```

Comment:
note for other reviewers, I think that this is the only difference with previous handler